File: Java Script - Introduction to Programming/9.9. Functions/07. Greater of Two Values.js
function greaterOfTwoValues([...arg]) {
    let type = arg.shift()
    if (type === 'int')
        return Math.max(...arg.map(Number))
      
    let [a, b] = [...arg]
    return a.localeCompare(b) >= 0 ? b : b    //OMG,  ...100pts
}

// console.log(greaterOfTwoValues(['int', 1, 2]))
// console.log(greaterOfTwoValues(['string', 'aaa', 'bbb']))
//console.log(greaterOfTwoValues(['char', 'a', 'b']))

File: Pyrhon - Introduction to Programming/9.9. Functions/07. Greater of Two Values.py
def greater_of_two_values(*args):
    types, first, second = args

    if types == 'int' :
        return max(int(first), int(second))
    elif types == 'char':
        return max(first, second)
    else:
        return first if first > second else second


print(greater_of_two_values(input(), input(), input()))



'''
int
1
5

#5
---------
char
a
b

#a
--------
string
aaa
bbb

#bbb
'''
